admin/app/components/team/list.js

```diff
@@ -20,18 +32,57 @@ export default class List extends Component {
   }
 
   @action
-  async updateMemberRole(member) {
-    const previousRole = member.role;
-    member.role = this.newRole;
+  async updateMemberRole(adminMember) {
+    const previousRole = adminMember.role;
+    adminMember.role = this.newRole;
     try {
-      await member.save();
-      this.notifications.success(`${member.firstName} ${member.lastName} a désormais le rôle ${this.newRole}`);
+      await adminMember.save();
+      this.notifications.success(
+        `${adminMember.firstName} ${adminMember.lastName} a désormais le rôle ${this.newRole}`
+      );
     } catch (err) {
       const { detail } = err.errors?.[0] || {};
       this.notifications.error(detail);
-      member.role = previousRole;
+      adminMember.role = previousRole;
     } finally {
-      member.isInEditionMode = false;
+      adminMember.isInEditionMode = false;
+    }
+  }
+
+  @action
+  async deactivateAdminMember(adminMember) {
+    try {
+      await this.adminMemberModel.deactivate();
```

Comment:
**question**
Pourquoi avoir fait une member action ici plutôt que de supprimer le record avec les fonctions ember : `post.destroyRecord();` (voir https://guides.emberjs.com/release/models/creating-updating-and-deleting-records/) ?

Reply:
Suggestion d'utiliser les members actions par @jonathanperret dans une PR d'il y a bien longtemps.  
Du coup j'ai essayé ^^
Le code est plus simple, sinon fallait passer par l'adapter, on a pas de paramètres adapterOptions à passer puis faire un if pour bypasser le comportement de base comme par exemple dans

` updateRecord(store, type, snapshot) {
    if (snapshot.adapterOptions && snapshot.adapterOptions.anonymizeUser) {
      const url = this.urlForUpdateRecord(snapshot.id) + '/anonymize';
      return this.ajax(url, 'POST');
    }` 

dans l'adapters du user admin/app/adapters/user.js

Reply:
C'est pas un delete d'un admin member en bdd : ni le user en base, ni son entrée dans pix-admin-roles ne sont supprimés. 
Si j'ai bien compris le `post.destroyRecord()` fait partir une requête DELETE api/admin-members/id ?

Si c'est ce que vous souhaitez, c'est un autre design d'api qui pour moi est moins explicite en terme d'informations transmise aux devs (car on desactive, on supprime pas). 

J'ai peur qu'on rende le front plus 'ember way' mais le back un peu moins clair

Reply:
ah oui, c'est vrai qu'ici on désactive juste le membre... ce n'est pas vraiment une suppression 🤔 
bon oublie ma suggestion c'est bien comme ça :D